### PR TITLE
Serve UI from api server

### LINKED ISSUE #6934: Serve the static files from `odo dev`
/kind user-story
/area dev

## User Story
As an odo user, I want to have a way to load any Devfile, visualize, edit, or save it, So that I can easily work with Devfile without having to understand it too much.

## Acceptance Criteria 
- [ ] It should integrate the [Devfile Builder PoC](https://github.com/feloy/devfile-builder) (currently reachable at https://devfile.odo.dev/) into odo
- [ ] It should serve the static files on the same port as the API server and at the root level (without `/api/v1`)
- [ ] This should be experimental

## Links 
- Related Epic (mandatory): https://github.com/redhat-developer/odo/issues/6895
- Requires the API server: https://github.com/redhat-developer/odo/issues/6301

NOTES:
- use of go embed
- Take care of how to build this and include the static files into the binary
- Separate tree of sources

/kind user-story

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this:**

/kind feature

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #6934

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**
- Static files for UI can be generated with the command `make ui-static`
- The files are not generated every time odo binary is built
- Instead, a Github Action is executed to check that these files are up-to-date
- A failing job is visible at: https://github.com/redhat-developer/odo/actions/runs/5473689088/jobs/9967493105
-

### CODE REVIEW

File: .github/workflows/check-generated-files.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Generate static files for UI
+        run: make ui-static
+
+      - name: Verify Changed files
+        uses: tj-actions/verify-changed-files@v16
```
Comment (rm3l) [inline .github/workflows/check-generated-files.yaml:18]: I think it is recommended to pin untrusted (i.e. not published by actions, docker, or redhat-actions) third-party actions to their commit SHA instead: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions
I remember an internal email we received a while back about this.
But I noticed this is not done consistently in all our Workflows. We can therefore do need consistently in a subsequent PR if you prefer..

**Review by rm3l — COMMENTED** [review]
Also, same point already mentioned before: can you add the generated `pkg/apiserver-impl/ui` to the `.gitattributes` file?

**Review by rm3l — APPROVED** [review]

**Comment by feloy** [comment]
/override ODO-PODMAN-TEST
```
 [FAILED] [91.208 seconds]
2023-07-07T09:47:22.5397480Z odo dev command tests when using devfile that contains K8s resource to run it on podman [It] should build the images when running odo dev on podman [podman]
2023-07-07T09:47:22.5398324Z /home/runner/work/odo/odo/tests/integration/cmd_dev_test.go:3710
[...]
2023-07-07T09:47:22.6078901Z   [odo] error starting container da1851b2f8e4d5a78ec3569ad8f4d627b96c694ba2d12366e5f1853558f05510: rootlessport listen tcp 127.0.0.1:24481: bind: address already in use
```

**Comment by feloy** [comment]
/override OpenShift-Integration-tests/OpenShift-Integration-tests
```
[FAILED] [123.518 seconds]
E2E Test starting with non-empty Directory test debugging [It] should verify developer workflow from non-empty Directory
/go/odo_1/tests/e2escenarios/e2e_test.go:326
```

**Pull request merged**